poke_redirect.py

```diff
@@ -0,0 +1,277 @@
+    ASM_Branch(True, addr + 40, config.fs_read),  # bl FSFILE_Read
+    b'\x30\x00\xbd\xe8',  # ldmia sp!, { r4,r5 }  ; pop variables
+    b'\x0d\x00\xa0\xe1',  # mov   r0, sp          ; set r0 to file handle
+    ASM_Branch(True, addr + 52, config.fs_close), # bl FSFILE_Close
+    b'\x3f\x80\xbd\xe8',  # ldmia sp!, { r0-r5, pc }
+    b'\x00\x00\x00\x00',  # nop
+  ))
+
+
+def ASM_WriteSDSaveFile(config: Config, addr: int, open_addr: int) -> bytes:
+  args = list(range(4))
+  if config.flip_write:
+    args[2], args[3] = args[3], args[2]
+  return b''.join((
+    b'\x7f\x40\x2d\xe9',  # stmdb sp!, { r0-r6, lr }
+    b'\x00\x60\xa0\xe3',  # mov   r6, #0x0        ; set flags 0 to r6
+    args[3].to_bytes(1, byteorder=CODE_BYTEORDER)
+      + b'\x50\xa0\xe1',  # mov   r5, r<args[3]>  ; set write size to r5
+    args[2].to_bytes(1, byteorder=CODE_BYTEORDER)
+      + b'\x40\xa0\xe1',  # mov   r4, r<args[2]>  ; set buffer pointer to r4
+    b'\x00\x30\xa0\xe3',  # mov   r3, #0x0        ; set offsetHigh as 0 to r3
+    b'\x01\x20\xa0\xe1',  # mov   r2, r1          ; set offsetLow to r2
+    b'\x0d\x10\xa0\xe1',  # mov   r1, sp          ; set r1 to sp to store file handle
+    ASM_Branch(True, addr + 28, open_addr),       # bl OpenSDSaveFile
+    b'\x01\x00\xa0\xe1',  # mov   r0, r1          ; set r0 to file handle
+    b'\x70\x00\x2d\xe9',  # stmdb sp!, { r4-r6 }  ; push variables
+    b'\x0c\x10\x8d\xe2',  # add   r1, sp, #0xc    ; set r1 to throwaway location for bytes written
+    ASM_Branch(True, addr + 44, config.fs_write), # bl FSFILE_Write
+    b'\x70\x00\xbd\xe8',  # ldmia sp!, { r4-r6 }  ; pop variables
+    b'\x0d\x00\xa0\xe1',  # mov   r0, sp          ; set r0 to file handle
+    ASM_Branch(True, addr + 56, config.fs_close), # bl FSFILE_Close
+    b'\x7f\x80\xbd\xe8',  # ldmia sp!, { r0-r6, pc }
+    b'\x00\x00\x00\x00',  # nop
+  ))
+
+
+def ASM_CheckSDSaveFile(config: Config, addr: int, open_addr: int) -> bytes:
+  return b''.join((
+    b'\x06\x40\x2d\xe9',  # stmdb sp!, { r1, r2, lr }
+    b'\x01\x00\x2d\xe9',  # stmdb sp!, { r0 }       ; push extra stack location for file handle
+    b'\x0d\x10\xa0\xe1',  # mov   r1, sp            ; set r1 to sp to store file handle
+    ASM_Branch(True, addr + 12, open_addr),         # bl OpenSDSaveFile
+    b'\xa0\x2f\xb0\xe1',  # movs  r2, r0, lsr #0x1f ; get the most significant bit of the result
+    b'\x01\x20\x22\xe2',  # eor   r2, r2, #0x1      ; flip the bit so true if success
+    b'\x0d\x00\xa0\xe1',  # mov   r0, sp            ; set r0 to file handle
+    ASM_Branch(True, addr + 28, config.fs_close),   # bl FSFILE_Close
+    b'\x01\x00\xbd\xe8',  # ldmia sp!, { r0 }       ; pop extra stack location
+    b'\x02\x00\xa0\xe1',  # mov   r0, r2            ; set r0 to result
+    b'\x06\x80\xbd\xe8',  # ldmia sp!, { r1, r2, pc }
+    b'\x00\x00\x00\x00',  # nop
+  ))
+
+
+def ASM_OpenSDSaveFile(config: Config, addr: int, save_path: str) -> bytes:
```

Comment:
Ah cool, a configurable path :tada:

Reply:
Ya, that was one of the main reasons for the script.  I'm actually working on a PKSM script that would just create the patch for you so users can do everything on a 3DS a lot more automatically.